editor/src/sample-projects/sample-project-utils.spec.ts

```diff
@@ -0,0 +1,9 @@
+import { defaultProject } from './sample-project-utils'
+import * as fs from 'fs'
+
+describe('Project model integrity check', () => {
```

Comment:
is this test here to help us commit the latest SampleProject.json to the repository? or is the idea that there's a committed SampleProject for fallback, but ideally this test will run before the server tests execute?

Reply:
The former. This reminds me though, I need to fix up the workflows that run on PR pushes to ensure that server tests are run on changes that affect the server, since right now we're only running editor tests on PR pushes.

wrt the ordering of the tests - the editor tests and server tests run in parallel as two separate jobs, meaning that even if the editor tests ran first the updated file wouldn't be used in the server tests since it wouldn't be committed to the repo by the CI. We can try to fix that but I think a better fix right now is to just add a check in this `.spec` that the new file matches the old one, then just overwrite if running locally or fail the test if running in the CI.

Reply:
hmm. could we instead make a `ts-node` script that is ran by the server tests to generate this file on-the-fly?

Reply:
I ended up spending 2 hours trying different solutions here and have given up for now. The problem is that we create the default project through code, and in a way that (inconsequential) parts of it are auto-generated and will change each time. That means having a test that checks if the file has changed will always break without a really brittle way of removing those parts, and that a `ts-node` script needs to pull in the editor project to get the dependencies it needs around generating a default `PersistentModel`. I tried adding one to our project but that fails to compile because it struggles to resolve the dependencies, and I also tried adding a separate webpack config to build an executable bundle for a script to generate that model, but had problems there too. I'll come back to this when I have a chance, but until then I have `xit`ed the test to save us from committing a change to that sample project JSON file with every run of the tests.

Reply:
oh, it's too bad `ts-node` couldn't handle the editor codebase! thank you for looking into this :)

Reply:
I think it most likely can, but needs some time spent figuring out the exact config required.